pandas/core/arrays/datetimes.py

```diff
@@ -152,7 +152,7 @@ class DatetimeArray(dtl.DatetimeLikeArrayMixin, dtl.TimelikeOps, dtl.DatelikeOps
 
     _typ = "datetimearray"
     _scalar_type = Timestamp
-    _recognized_scalars = (datetime, np.datetime64)
+    _recognized_scalars = (datetime, np.datetime64, date)
```

Comment:
this is the right way to address #35466, but you also need to look at the other places where _recognized_scalars is used and make sure those tests are updated appropriately